File: scripts/render.py
import sys
import os
import jinja2
import yaml

def render(tpl_path, context):
    path, filename = os.path.split(tpl_path)
    
    with open(tpl_path, 'r') as f:
        data = '{% extends "template.j2" %}' + f.read()
    return jinja2.Environment(
        loader=jinja2.FileSystemLoader(path or './')
    ).from_string(data).render(context)
    
with open(os.path.join(sys.path[0], "../src/template/template_data.yaml"), 'r') as f:
    data = yaml.load_all(f.read())
context = next(data)

result = render(os.path.join(sys.path[0], "../src/template/.index.html"), context)

with open(os.path.join(sys.path[0], "../src/index.html"), 'w') as f:
    f.write(result)

File: scripts/render.js
const yaml = require('js-yaml');
const fs = require('fs');
require('jinjs').registerExtension('.j2', (txt, x) => `{% extends "./template.tpl" %}${txt}`);
require('jinjs').registerExtension('.tpl');

const context = yaml.safeLoadAll(fs.readFileSync('src/template/template_data.yaml', 'utf8'))[0];
const myTemplate = require('../src/template/index.j2');

const result = myTemplate.render(context);
const outFile = 'src/template/index.html';
fs.writeFile(outFile, result, (err) => {
    if (err) {
        console.log(err);
    }
    else {
        console.log(`${outFile} was successfully saved.`);
    }
});
